### ISSUE TITLE
Editing Feed Posts

### ISSUE THREAD

**Issue by Cerno-b** [ludumdare/ludumdare#992]
When I edit a Feed post, I can only save the change, not actually go to the post. I can preview, edit and save, but there is no way to return to the actual post view that contains the comments.

A workaround it to remove the /edit in the address or use the browser's back button, but a specific "go to post" button would be nicer.

This is also true for the entry page.

**Comment by local-minimum** [comment]
If I understand you correctly you are sort of referring to an `Abort` kind of button. We discussed this a while back but I think we forgot to implement a solution. Got stuck discussing suitable word for the action.

**Comment by Cerno-b** [comment]
No, not abort. More like "Save and Close"

**Comment by local-minimum** [comment]
Hmm yeah that sounds like a bug

**Comment by local-minimum** [comment]
In my local dev-environment I can see that going into edit-mode and not making any changes makes save button behave differently than if I make changes. If I make changes state swaps correctly out of edit-mode thanks to hack I did. But I need to tweak the URI too so that it exits the edit completely.

**Comment by local-minimum** [comment]
I think the interaction pattern was designed that the `Save` should not mean "I am done", but more "I don't want to loose my progress". I'm not really happy with how it works now though with the `Saved` button looking very intractable while it does nothing.

If you ask me, we need `Post` on the left side after `Edit` for going out of edit mode. But the word `Post` is bad because it sounds like you are posting while it would indeed be aborting if you haven't saved. It could ask if you want to save before going to `Post`. Reason I want it on left side because it is a view-mode rather than an action.

**Comment by Cerno-b** [comment]
I don't think I follow. You currently have Edit and Preview on the left, Save on the right.

I understand that you want to switch between the Editor and the Preview and that's what the buttons on the left do, quite clear. Then you have Save on the right which updates your post, also very clear to follow. What is missing though is the option to leave the Edit view and return to the Post view.

There would be two ways of leaving the Edit view: "Save and Exit" and "Discard and Exit". I would argue that "Discard and Exit" is not something people will often use and if they do, they could just as well close the tab. On the other hand "Save and Exit" is something people will do all the time, if they want to see the final version of the post.

Having the buttons "Save", "Save and Exit", "Discard and Exit" on the right seems perfectly fine to me, as it visually splits two different types of buttons. On the left you switch between modes but don't change the content, on the right you decide about changing (or not changing) the content.

By the way, is there a button type that makes it clearer that Preview and Edit are exclusive? Like the Android menu thingy where a slider goes to the left or the right? I think that would make the way these buttons work a bit more intuitive.

**Comment by Cerno-b** [comment]
By the way, what you described above does not happen for me on the site (Firefox). 

If I open an existing post, it says "Saved", but I can click the button (hovering turns it red, clicking turns it permanently red until I click another element). This is a bit weird. I would expect the "Saved" button to be completely inactive. No reaction to hovering, no reaction to clicking.

Now if I change the test, the button turns to a blue "Save" button, and when I click it, it turns to the weird "Saved" button behavior. 

In neither case does it return to the post view. 

So if you want the Save button to return to the Post, then that's broken (for me)
If staying in the Edit view was intentional, then I would recommend adding a "Save and Close" button.

**Comment by local-minimum** [comment]
I agree with most everything you say and I was a bit confused about what were things I had modified locally.

I agree `Saved` is weird / shouldn't look like a button.

Problem: Mike will want one short word for each action and must work worth only icon too because need to be mobile friendly.

The view mode buttons do appear throughout the site but seldom only two. Somhow they are less clear here perhaps. In general, the `Edit`-bar with all the buttons could need reworking. We'll get dropdowns soon, so we'll see then.

**Comment by Cerno-b** [comment]
I can understand these design decisions. But should the default behavior of Save not be "Save and Close"? That would solve my problem and the button would keep its short name.

Maybe it's just me, but I think if you modify an existing post you often only make small changes and don't have the need for making intermediate saves after which you continue typing. And if you do, it's as easy as pushing the browser's back button. I just feel that the more common action should get the better control flow.

**Comment by povrazor** [comment]
Loosely related to #649
